apps/web/src/hooks/v3/usePools.ts

```diff
@@ -190,58 +190,63 @@ export function usePoolsWithMultiChains(
   }, [poolTokens])
 
   const chainIds = useMemo(() => Object.keys(poolAddressMap), [poolAddressMap])
+  const poolAddressesString = useMemo(
+    () => chainIds.flatMap((chainId) => poolAddressMap[chainId]).join(','),
+    [chainIds, poolAddressMap],
+  )
 
-  const queries = chainIds.map((chainId) => {
-    const poolAddresses: Address[] = poolAddressMap[chainId]
-    return {
-      queryKey: ['slot0', 'liquidity', chainId, ...poolAddresses],
-      placeholderData: keepPreviousData,
-      queryFn: async () => {
-        const client = publicClient({ chainId: Number(chainId) })
-
-        const slot0Calls = poolAddresses.map((addr) => ({
-          address: addr,
-          abi: v3PoolStateABI,
-          functionName: 'slot0',
-        }))
-
-        const liquidityCalls = poolAddresses.map((addr) => ({
-          address: addr,
-          abi: v3PoolStateABI,
-          functionName: 'liquidity',
-        }))
-
-        return Promise.all([
-          client.multicall({
-            contracts: slot0Calls,
-            allowFailure: false,
-          }),
-          client.multicall({
-            contracts: liquidityCalls,
-            allowFailure: false,
-          }),
-        ])
-      },
-      enabled: !!chainId && !!poolAddresses.length,
-      ...QUERY_SETTINGS_IMMUTABLE,
-      ...QUERY_SETTINGS_WITHOUT_INTERVAL_REFETCH,
-    }
-  })
-  const poolInfoQueries = useQueries({
-    queries,
+  const { data: poolInfoByChainId } = useQuery({
```

Comment:
the problem is useQueries will cause re-render when every query update?

Reply:
I think useQueries can control of cache more precise, for example, if there is one poolAddress update, queries can just fetch only one chain's data, but if use single query, all chainIds need to refetch again. so it's a tradeoff